### ISSUE TITLE
Windows Property Variations

### ISSUE THREAD

**Issue by ermau** [xamarin/Xamarin.PropertyEditing#95]
- Core (#94)
 - Experience:
   - [ ] Design
   - [ ] Localized
   - Behavior:
     - [ ] Keyboard control
     - [ ] Accessibility
   - Theming:
     - [ ] Light themed
     - [ ] Dark themed
     - [ ] High contrast

**Comment by vancura** [comment]
Very similar to https://github.com/xamarin/Xamarin.PropertyEditing/issues/96, here's a Windows version:

What do you think about the following flow?

1. Standard view: some properties support variations, but none are defined.

![1 2x](https://user-images.githubusercontent.com/4982/35049381-1cf7beb6-fba0-11e7-80cf-4d6b5eff0e16.png)

2. Considering the user wants to add a variation, they rollover above the _Add a customization_ button. We show a tooltip, similar to Xcode; we might need to reword this to match the rest of UI, I have copied the text from there. The click area of the button needs to be larger than the glyph, so it is easy to hover and click (preferably via a standard `16pt` icon).

![2 2x](https://user-images.githubusercontent.com/4982/35049421-3e24a77a-fba0-11e7-9563-563684d2687c.png)

3. The user clicks the _Add a customization_ button and we show them this dialog:

![3 2x](https://user-images.githubusercontent.com/4982/35049451-5b7c857c-fba0-11e7-9d21-502131fbc3a5.png)

4. This is a small improvement from Xcode: if the variation settings don't provide meaningful results, we disable the button:

![4 2x](https://user-images.githubusercontent.com/4982/35049494-7db970fa-fba0-11e7-9235-4f05b1c81446.png)

5. The user added some variations. Mainly for beginners, it might be tricky to understand the shorthands (_wR_, _hC_, _gP_ and so on), so we enable tooltips with explanations. This is the second improvement from Xcode; they do not explain anything.

![5 2x](https://user-images.githubusercontent.com/4982/35049533-9dd89096-fba0-11e7-88f8-ffaf040df76f.png)

6. When a variation is activated via variations panel (resulting in a preview of that variation), we highlight that variation with a color; we might need to add another visual treatment for better a11y (design pending).

![6 2x](https://user-images.githubusercontent.com/4982/35049578-bc12c39c-fba0-11e7-956c-0df11f477a90.png)

7. The _Remove a variation_ button removes the variation from the list. We display a tooltip here too.

![7 2x](https://user-images.githubusercontent.com/4982/35049593-cd7c63cc-fba0-11e7-872b-8723989848fb.png)

Thoughts?

**Comment by vancura** [comment]
Light vs. Dark IDE themes:

<img width="408" alt="win variations active 2x" src="https://user-images.githubusercontent.com/4982/35334660-6585dcbc-0113-11e8-8dee-8f1b032a3922.png">

<img width="408" alt="win variations active dark 2x" src="https://user-images.githubusercontent.com/4982/35334663-67bb327a-0113-11e8-8996-77dc5eb39b47.png">

**Comment by ermau** [comment]
@vancura Can we revisit this with the perspective of not purely the constraints that iOS defines? I don't know that single letter abbreviation work in a more dynamic context.

**Comment by vancura** [comment]
@ermau Sure, do we have a list of potential values?

**Comment by ermau** [comment]
@vancura No, that's sort of the point, it needs to be a bit more open ended.

**Comment by ermau** [comment]
Additionally what does this look like when within a grouped editor?

**Comment by ermau** [comment]
@vancura It looks identical to the WPF screenshots. As far as multi-selection state, it depends on the editor and the state of values. If all selected objects have the same value for the property, it looks no different than normal. If they have disagreeing values, it has a blank or indeterminate state.

**Comment by ermau** [comment]
To add: If multiple objects have the same variant for a property but different values, we can handle that in the same way we do now for a property. I'm mainly concerned with if one has a variant and the other doesn't. We could just treat it as if they have disagreeing values and add the variant where it's missing if they set the value.
